File: src/main/java/com/example/demo/user/profile/dto/UpdateUserProfileRequest.java
package com.example.demo.user.profile.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public class UpdateUserProfileRequest {

    private boolean nicknameProvided;
    private String nickname;

    public boolean nicknameProvided() {
        return nicknameProvided;
    }

    public String nickname() {
        return nickname;
    }

    @JsonProperty("nickname")
    public void setNickname(String nickname) {
        this.nicknameProvided = true;
        this.nickname = nickname;
    }
}

File: src/main/java/com/example/demo/user/profile/dto/UpdateUserProfileRequest.kt
package com.example.demo.user.profile.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
class UpdateUserProfileRequest {
    private var nicknameProvided = false
    private var nickname: String? = null

    fun nicknameProvided(): Boolean = nicknameProvided

    fun nickname(): String? = nickname

    @JsonProperty("nickname")
    fun setNickname(nickname: String?) {
        this.nicknameProvided = true
        this.nickname = nickname
    }
}
